Source/CarthageKit/Project.swift

```diff
@@ -371,7 +371,42 @@ public final class Project { // swiftlint:disable:this type_body_length
 					}
 					.map { $0.0.name }
 					.collect()
+		}
+	}
+
+	/// Finds the required dependencies and their corresponding version specifiers for each dependency in Cartfile.resolved.
+	func requirementsByDependency(
+		resolvedCartfile: ResolvedCartfile,
+		tryCheckoutDirectory: Bool
+	) -> SignalProducer<CompatibilityInfo.Requirements, CarthageError> {
+		return SignalProducer(value: resolvedCartfile)
+			.map { resolvedCartfile -> [(Dependency, PinnedVersion)] in
+				return Array(resolvedCartfile.dependencies)
```

Comment:
Instead of `map` and `flatMap`, this can create a `SignalProducer` directly:

```swift
return SignalProducer(resolvedCartfile.dependencies)
  .flatMap(.concurrent(limit: 4)){ dependencyAndPinnedVersion ->
```

This is important because we have the `concurrent` in the wrong spot right now. (That's probably my fault. Sorry!)